pkg/agent/route/interfaces.go

```diff
@@ -51,6 +52,24 @@ type Interface interface {
 	// DeleteSNATRule should delete rule to SNAT outgoing traffic with the mark.
 	DeleteSNATRule(mark uint32) error
 
+	// InitService should add the basic TC configuration on Linux.
+	InitService(nodePortIPMap map[int][]net.IP, isIPv6 bool) error
+
+	// AddNodePort should add related configuration about the NodePort Service to TC on Linux.
+	AddNodePort(nodePortIPMap map[int][]net.IP, port uint16, protocol binding.Protocol) error
+
+	// DeleteNodePort should delete related configuration about the NodePort Service to TC on Linux.
+	DeleteNodePort(nodePortIPMap map[int][]net.IP, port uint16, protocol binding.Protocol) error
+
+	// AddClusterIPRoute should add route on k8s node for Service ClusterIP.
+	AddClusterIPRoute(svcIP net.IP, isIPv6 bool) error
```

Comment:
To be consistent with other funcs, I suggest either add "Routes" to other func names (I know they are TC operations, but that needs not be reflected in the RouteClient interface), or remove "Route" from this func name too.

Probably Route -> Routes, as I assume it might not be a single route entry?

Reply:
Why there is not a func to remove the routes for a ClusterIP Service?

Reply:
> To be consistent with other funcs, I suggest either add "Routes" to other func names (I know they are TC operations, but that needs not be reflected in the RouteClient interface), or remove "Route" from this func name too.
Probably Route -> Routes, as I assume it might not be a single route entry?

I don't know if adding the suffix "Routes" to func name will cause misunderstood.  

-  For funcs `AddNodePort`/`DeleteNodePort`, no route entry will be added/deleted, only TC rules.  
- For funcs `AddLoadBalancer`/`DeleteLoadBalancer`, route entries and TC rules are both added/deleted.
- For func `AddClusterIPRoute`,  only route entry will be added or updated. 

I don't know if "Route" is an appropriate suffix for all the above funcs.

> Why there is not a func to remove the routes for a ClusterIP Service?

We don't a func to remove the route for a ClusterIP Service. There is alway one route entry for all ClusterIP Services. 

- If there is no route entry when a ClusterIP is added, create a route entry for it.
- If current route entry can route the ClusterIP, there is nothing to do.
- If current route cannot route the ClusterIP, replace the current route with a new route entry , and the new route entry 's destination IP block can route the ClusterIP IP.

Reply:
But how if the ClusterIP CIDR is changed and a ClusterIP is no more valid?

Reply:
That's no matter.  What we wanted is to calculate the IP block of ClusterIP finally as possible, and this IP block is the destination of the only one routing entry for ClusterIP. Adding a ClusterIP may help enlarge the destination IP block(ClusterIP block) of routing entry.
Delete a ClusterIP is ok, as we want just only one routing entry which is used to route ClusterIP traffic. If we add a ClusterIP, then we create a routing entry for it, and when it is deleted, we delete the routing entry,  then there will be lots of routing entries.

Reply:
Ok. Then if we assume CIDR can only be extended we should be ok.

Could you add some comments before the AddClusterRouteIP() implementation in route_linux.go to explain what it does? Actually good to add comments before other RouteClient funcs  you added too.

Reply:
Thanks, I'll add some comments.